File: Commander/WebView.cs
using WebWindowNetCore;

class WebView
{
    public static DateTime StartTime { get; } = DateTime.Now;

    public static WebWindow Window { get; private set; } = null!;

    public static void Run()
    {
        Window = WebWindow
            .Builder()
            .AppId(Globals.APP_ID)
            .Title("Commander")
            .WithDiagnostics(true)
            .InitialBounds(600, 800)
            .SaveBounds()
            .DevTools()
            .DefaultContextMenuDisabled()
#if Windows
            .OnCreating(Form.OnCreate)
            .ResourceIcon("icon")
            .WithoutNativeTitlebar()
            .OnStateChange(w => Requests.SendJson(new(null, EventCmd.WindowState, new EventData { Maximized = w.IsMaximized })))
#else
            .FromResourceTemplate("template", Commander.Platform.Linux.Window.OnActivation, true)
#endif
            .DebugUrl($"http://localhost:5173/#platform={Globals.Platform}")
            .Url($"http://localhost:8080##platform={Globals.Platform}")
            .CanClose(BackgroundJobs.IsIdle)
            .Build();
        Window.Run();
    }
}

static class WebViewExtensions
{
#if Linux
    public static Commander.Platform.Linux.Window GetNative(this WebWindow ww) => (ww.Window as Commander.Platform.Linux.Window)!;
#endif
}

File: Commander/WebView.fs
module WebView

open System
open System.Drawing
open WebWindowNetCore
open Requests
open Elevated

let run () = 
    WebView()
        .AppId("de.uriegel.commander")
        .InitialBounds(600, 800)
        .Title("Commander")
        .BackgroundColor(Color.Transparent)
        .WithoutNativeTitlebar()
        .ResourceIcon("icon")
        .ResourceFromHttp()
        .DebugUrl("http://localhost:5173")
        .CorsDomains([|"http://localhost:5173"|])
        .CorsCache(TimeSpan.FromSeconds(20))    
        .SaveBounds()
        .DefaultContextMenuDisabled()
        .AddRequest("getroot", Root.get)
        .AddRequest("getfiles", Directory.getFiles)
        .AddRequest("gettrackinfo", getTrackInfo)
        .AddRequest("getextendeditems", Directory.getExtendedInfos)
        .AddRequest("cancelextendeditems", Directory.cancelExtendedInfos)
        .AddRequest("onenter", Directory.onEnter)
        .AddRequest("onshowdir", Directory.onEnter)
        .AddRequest("renameitem", tryElevatedOnAccessDenied Directory.renameItem)
        .AddRequest("deleteitems", tryElevatedOnAccessDenied Directory.deleteItems)
        .Requests([getIcon; getFile])
        .OnEventSink(
            fun id webview -> Events.onEventSink id (
                fun d -> webview.SendEvent.Invoke(id, d)
            )
        )
#if DEBUG    
        .DevTools()
#endif
#if Linux
        .TitleBar(Titlebar.create)
#endif    
        .Run()
